subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/capabilities/CapabilitiesLocalComponentIntegrationTest.groovy

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.capabilities
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class CapabilitiesLocalComponentIntegrationTest extends AbstractIntegrationSpec {
+
+    def "can detect conflict between local projects providing the same capability"() {
+        given:
+        settingsFile << """
+            rootProject.name = 'test'
+            include 'b'
+        """
+        buildFile << """
+            apply plugin: 'java-library'
+            
+            capabilities {
+                api 'org:capability:1.0'
+            }
```

Comment:
I'm not really fond of this, particularly due to the infrastructure required to make it work nicely in Groovy and Kotlin.
Instead, I'd add a `capability` method to the `ConfigurationPublications` API, as Adam (kind of) suggested.

We could later move all of the `outgoing` stuff into a better place, but I believe it belongs together.

Reply:
Ok, actually I didn't know there was such a thing already. `TooManyNewAPIsException`.